brainframe_client.py

```diff
@@ -3,7 +3,6 @@
 from brainframe_qt.util import preimport_hooks
```

Comment:
(Code line is irrelevent, I dont know any other way to have threaded conversations)

**Problem**: Closing the client hangs forever

This doesn't seem to be a regression (it is possible to replicate on v0.27), but it is now critical to fix because it prevents the opening of another client.

**Replicate**: 

1) Run the client without a server running
2) Close the client splashscreen
3) Client never closes (and it's impossible to start another client because of this)
4) The only way to close the client is to CTRL+C the terminal, which would be bad if the user opened it through the .desktop

Reply:
How are you closing the splashscreen w/o Ctrl+C-ing the program? Alt-F4?

Reply:
I have "Super+Q" mapped to "Close Window". That's irrelevant though, I just tested it and right clicking on the taskbar and clicking "Close" has the same effect. 

I was able to reproduce this on Windows, Linux, and the snap.

Reply:
Ok, so using the window manager to close it, gotcha. There's no close button on the splashscreen, which is why I was asking. Will investigate.

Reply:
Is this a new issue with this PR? Does this not occur on `main` branch?

Reply:
> This doesn't seem to be a regression (it is possible to replicate on v0.27), but it is now critical to fix because it prevents the opening of another client.

Reply:
The issue is if you open a client, then close it in the splashscreen, you will never be able to open the client again.

Reply:
Didn't you find out the problem was the ZoneStatusPoller thread not being a daemon?

Reply:
No, that was the cause of a different problem :sweat_smile:. That issue prevented the client from closing after the API was already talking to the server

Reply:
I think I know _what_ causes this problem. 
https://github.com/aotuai/brainframe-qt/blob/e97f0c5f4835509641971ded91ac909f6516f194/brainframe_qt/ui/brainframe_app.py#L243-L247

The client gets stuck here waiting for an event, not realizing the QApplication is closing down around it

Reply:
I guess you can add another event there that is triggered when the QApplication closes? :thinking:

Reply:
See #139